### DISCUSSION TITLE
Codespace prebuild is wrongly accumulating storage

### DISCUSSION THREAD

**Discussion by stauffenberg2020** [community/community#49291] [Codespaces] [answered]
### Select Topic Area

Bug

### Body

Hi All,

I have disabled my codespace prebuilds and deleted all the old codespace prebuild actions runs couple of days ago in my repo. I am surprised why my consumption continuously starts to increase even though I am not using any such services. Because of this, I am unable use Codespace because Github says that I have used 100% of my codespace usage (through wrongly billed Codespace storage)

It looks like this is a bug similar to 
https://github.com/orgs/community/discussions/48213#discussioncomment-5081863
and
https://github.com/orgs/community/discussions/48457

Is there any way this bug can be overcome without Github staff looking into it from the backend when they have time?

Thanks in advance

**Comment by james-ecd** [comment]


**Reply by stauffenberg2020** [reply]
Hi James, 

Thanks for your reply. I have already checked steps 1 and 2 as per your recommendation. It didnt help me and thats why I am reaching out to Github's support for further help

**Reply by stauffenberg2020** [reply]
I am sorry I cannot do that. Because whatever you have posted is general guidance and is irrelevant to my question where I have already explained in detail and pointed to similar bug reports. I dont want to entertain this sort of Github community profile building technique

**Comment by KartikM123** [comment, marked as answer]
Hi @stauffenberg-2020 ! Thanks for reaching out!

After looking into our backend stores, we noticed there was a bug when you deleted your prebuild configuration. We went ahead and deleted those stale records. 

You shouldn't receive any more bills for those prebuilds; please don't hesitate to reach out if you don't see this fixed! Let me know if you have any questions and thanks again for your patience!

**Reply by jmduper** [reply]
What could I do to solve this?

**Reply by shuitest** [reply]
It seems the bug wasn't fixed in programming but in data. So it's resolved for the specific user but it happened again for the same use case.

The bug in detail:
If you have Prebuild Storage used somewhere, it's counted as the total codespace usage, somehow it automatically accumulates to reach the free quota so you can't use codespace any more. When you delete all your prebuild storage and all your codespaces, it didn't update your free quota restrictions. So you still can't use codespace and your displayed storage limit is still high. 

I didn't test this for paid users. If the codebase for paid and free layer are the same, the issue possibly exist for paid users too. 

The related links

The place to show the storage usage
https://github.com/settings/billing

![image](https://user-images.githubusercontent.com/5940220/236598362-f913e75e-ca49-446f-97aa-f02642414cf6.png)

**Reply by paulvantuyl** [reply]
Happened to me also. 31 m of prebuilt processing for a gitignore file when I wasn't using the codespace.

**Reply by nelisjunior** [reply]
I did [this here](https://github.com/orgs/community/discussions/49291#discussioncomment-7469089) and worked!

**Comment by clydedsouza** [comment]
Hi,
I have the same issue. I can see no Codespaces listed on my profile but I keep getting the storage alert email. In the usage information, I can see all prebuild storage being accumulated, as per screenshot below. Could you please assist? Thanks!

![image](https://user-images.githubusercontent.com/16275549/236671616-f3ad152b-0e08-43ad-b02f-de2cee72559a.png)

**Reply by mishra-prajjawal** [reply]
Hello @clydedsouza 
You May read the below [message](https://github.com/orgs/community/discussions/49291#discussioncomment-5840075).
Hope it helps
Thank you for contacting.

**Reply by clydedsouza** [reply]
Hey @mishra-prajjawal, thanks for that message, it was helpful. Out of the 4 repo's that I'd setup codespaces on, only 1 had a prebuild configuration set on 4 regions. I don't really even need a prebuild config, so I deleted it. I'm hoping that deleting it will also reduce the space?

**Reply by mishra-prajjawal** [reply]
> Hey @mishra-prajjawal, thanks for that message, it was helpful. Out of the 4 repo's that I'd setup codespaces on, only 1 had a prebuild configuration set on 4 regions. I don't really even need a prebuild config, so I deleted it. I'm hoping that deleting it will also reduce the space?

Hello @clydedsouza ,
Glad I could be of help,
Yes deleting the prebuilds will certainly reduce your codespace storage billing. 
I am adding an extract from GitHub  Docs and tagging a link to how the storage usage works: 
>The GB-month unit of storage is a time-based measurement, 1 GB-month being 1 GB of storage usage for one whole month. The disk space used by all of your codespaces and prebuilds is assessed once an hour and your current GB-month usage is recalculated. Therefore, while you have codespaces and prebuilds, your GB-month usage will increase throughout the month. For example, if the storage totals 15 GB, and remains unchanged throughout your monthly billing cycle, then you will have used 7.5 GB halfway through the month, and 15 GB at the end of the month. For more information, see "[About billing for storage usage](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces#about-billing-for-storage-usage)" later in this article. 

Link to the article : [About billing for GitHub Codespaces -> Monthly included storage and core hours for personal accounts](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces#monthly-included-storage-and-core-hours-for-personal-accounts)

---

Looking at what you said I would suggest to treat codespaces as ephemeral machines, Spin up a new one for every task and delete them after the task because even if the codespace is not running it is utilizing the storage consumption quota, Hence increasing the storage usage. 
Here is an article on how to get the best out of your GitHub Codespaces
[Getting the most out of your included usage](https://docs.github.com/en/codespaces/troubleshooting/troubleshooting-included-usage)

Thanks you again for contacting

**Comment by mishra-prajjawal** [comment]
Hello @stauffenberg-2020 

Sorry for the problem you are facing, 
based on the data you are sharing the problem could be associated to the only factor:
There could be Multiple **Region availability** in Prebuilds/New configuration of the repo you have made which would cause multiple regions to store your prebuild and increasing your prebuilds storage at an exponential rate.

--- 
**To solve the problem**<br>
1. Go to the repository where you had set prebuilds
2. Navigate to the repository settings  `Settings`
3. Under Settings menu select codespaces `Codespaces`
4. Check for **Region availability** untick the `Reduce prebuild available to only specific regions`  and remove the configuration
--- 
Hope it will help you, 
This is a bug from GitHub which should be a method to reduce time for opening a codespace and to manage the number of prebuilds .
You may read the documentation from GitHub to have a better understanding on how prebuilds work. 
<br>
[Configuring prebuilds](https://docs.github.com/en/codespaces/prebuilding-your-codespaces/configuring-prebuilds)
<br>
Thank you for contacting

**Comment by Silverium** [comment]
Hello, my account has the same issue, and I already deleted the codespace months ago. I can't do anything to remove the `prebuild storage`. Could anyone help me to fix that?

**Reply by martinlicht** [reply]
I am having the same issue. I have got some actions run every day, and I suspect their daily memory usage gets accumulated. There are no prebuilds or codespaces at all whatsoever.

**Reply by spenserblack** [reply]
Go to github.com/settings/billing and generate a usage report if you think you're wrongly getting usage. It should tell you what GitHub thinks you're using for each repository. Then you can investigate if those repositories have prebuilds set up.

**Reply by cgtarmenta** [reply]
@spenserblack  I did exactly that
the report:
![image](https://github.com/community/community/assets/10125168/9ebafb64-8e54-4d41-ab52-b8875279c766)

repository>settings>codespaces
![image](https://github.com/community/community/assets/10125168/9efc3e89-228c-4913-bfed-350f34df5173)

codespace usage:
![image](https://github.com/community/community/assets/10125168/be4bef21-c72d-4499-bf53-188d8740e54b)

I just tested codespaces around 3 days,  months ago. After my testing, I've removed all, and still, the prebuild is growing.

@KartikM123 please, can you remove all of those prebuilds, and stop them from continuing?

**Reply by spenserblack** [reply]
@cgtarmenta I had that exact same problem as you, actually. Prebuilds showed up on my report even though they were disabled. I opened up a discussion reporting this as a bug, and someone quickly responded to manually delete the prebuild (or whatever needed to happen on GitHub's side). After that, no more prebuild usage :smile:

**Reply by cgtarmenta** [reply]
@spenserblack Thanks, ill do that.

**Reply by nelisjunior** [reply]
I also had this prebuild problem with.
Your suggestion allowed me to come up with an idea.

1. I downloaded the suggest .csv file and identified that the problematic repository was one of  my fork  called "**web-latex**".
2. I acessed this page https://github.com/nelisjunior/web-latex/settings/codespaces and delete the prebuild codespace of this repository.
3. So i removed like that: <br><img src="https://docs.github.com/assets/cb-36637/mw-1440/images/help/codespaces/prebuilds-delete.webp" width="70%" height="70%">


---
**This is a generic url:** https://github.com/{username}/{repository}/settings/codespaces

**Comment by Silverium** [comment]
Spenser, I can’t find how to generate that. The current ongoing report does
not say which project. The codespace projects is completely empty. Months
ago

**Reply by spenserblack** [reply]
There should be a button in  that page called "get usage report". It is next to the "Usage this month" header.

**Comment by Silverium** [comment]
thanks, I managed to disable the prebuilds on that project that was
draining the space, even after having removed it from the codespaces
feature.

**Comment by karimkawambwa** [comment]
@KartikM123 I am having the same issue, could you also help resolve this for me.

Thanks.

![Screen Shot 2023-07-30 at 1 45 42 PM](https://github.com/community/community/assets/4308339/78506669-f594-4959-a070-919abc69a278)

**Comment by BrookRemington** [comment]
@KartikM123 I am also accumulating prebuild storage charges, even though I've deleted the the associated repository. Can you help fix this?

Thanks.

**Comment by daeh** [comment]
I think I might also be having related issues. The Usage report says that for 2 repos, a prebuild is running every day. One repo is deleted and the other repo doesn't have a prebuild configured. Any idea how I can get help with this?

**Reply by daeh** [reply]
This is still happening. Prebuilds are running every day, one on a deleted repo and another on a repo that shouldn't be running a prebuild (the prebuild is configured to run every push and I haven't pushed to that repo for over a year). @KartikM123 Can you check if there are stale records that can be deleted? Thanks much.

![CleanShot 2023-11-15 at 16 01 20](https://github.com/community/community/assets/12652166/74a4e44c-8391-4c01-a31c-99988a89fd81)


![CleanShot 2023-11-15 at 16 05 18](https://github.com/community/community/assets/12652166/176808c3-2d7d-4948-b4ac-138e79d93c9a)

**Reply by Silverium** [reply]
you did not push, but maybe dependabot did

**Reply by daeh** [reply]
I'd rather start with the **deleted repositories** issue. Clearly this isn't normal behavior. And WRT dependabot, I have other codespaces with the same specs as the named repo for which the prebuild is not running literally every single day. So no.

**Comment by forvaidya** [comment]
How to list repositores which have an active prebuild.
I also have the same problem - 
```
Storage
15.00 of 15.00 included GB-month used
```

**Reply by forvaidya** [reply]
actually I have no codespace 
```
mvaidya@AWHRVA bin % ./gh api /user/codespaces
{
  "codespaces": [],
  "total_count": 0
}
```

**Comment by Hugsoft** [comment]
I have same issue, i did not use codespace bit still showing pre build storage is full. Please help

**Reply by nelisjunior** [reply]
Try this: https://github.com/orgs/community/discussions/49291#discussioncomment-7469089

**Comment by rahulmr** [comment]
How to know which repository is the culprit? I do not have any codespace running at the moment.

**Reply by spenserblack** [reply]
From your settings page, go to billing. On that page, there should be an option to generate a usage report.
